examples/async_insert_without_waiting.ts

```diff
@@ -0,0 +1,126 @@
+import { createClient } from '@clickhouse/client'
+import { ClickHouseError } from '@clickhouse/client-common'
+import { EventEmitter } from 'events'
+
+// This example demonstrates how to use async inserts without waiting for an ack about a successfully written batch.
+// Run it for some time and observe the number of rows sent and the number of rows written to the table.
+// Suitable for ClickHouse Cloud, too. Can be used with either Node.js or Web versions of the client.
+// A bit more advanced version of the `examples/async_insert.ts` example,
+// as async inserts are an interesting option when working with event listeners
+// that can receive an arbitrarily large or small amount of data at various times.
+// See https://clickhouse.com/docs/en/optimize/asynchronous-inserts
+void (async () => {
+  const client = createClient({
+    host: process.env['CLICKHOUSE_HOST'], // defaults to 'http://localhost:8123'
+    password: process.env['CLICKHOUSE_PASSWORD'], // defaults to an empty string
+    max_open_connections: 10,
+    clickhouse_settings: {
+      // https://clickhouse.com/docs/en/operations/settings/settings#async-insert
+      async_insert: 1,
+      // https://clickhouse.com/docs/en/operations/settings/settings#wait-for-async-insert
+      // explicitly disable it on the client side;
+      // insert operations promises will be resolved as soon as the request itself was processed on the server.
+      wait_for_async_insert: 0,
+      // https://clickhouse.com/docs/en/operations/settings/settings#async-insert-max-data-size
+      async_insert_max_data_size: '1000000',
+      // https://clickhouse.com/docs/en/operations/settings/settings#async-insert-busy-timeout-ms
+      async_insert_busy_timeout_ms: 1000,
+    },
+  })
+  const tableName = 'async_insert_without_waiting'
+  await client.command({
+    query: `DROP TABLE IF EXISTS ${tableName}`,
+  })
+  await client.command({
+    query: `
+      CREATE TABLE ${tableName}
+      (id Int32, name String)
+      ENGINE MergeTree()
+      ORDER BY (id)
+    `,
+  })
+
+  type Row = { id: number; name: string }
+
+  // Assume we have an event listener in our application that periodically receives incoming data,
+  // that we would like to have inserted into ClickHouse.
+  // This emitter is just a simulation for the sake of this example.
```

Comment:
the code can just call `void asyncInsertOnData(rows)`?

Reply:
You might be right. This emitter was written just for the sake of the example.